File: Plugins/AudioPlayer/src/android/app/src/main/java/com/shortsands/audioplayer/AudioTOCBook.java
package com.shortsands.audioplayer;

import android.database.Cursor;
import android.util.Log;

/**
 * Created by garygriswold on 8/30/17.
 */

class AudioTOCBook {

    private static final String TAG = "AudioTOCBook";

    AudioTOCTestament testament;
    String bookId;
    String bookOrder;
    Integer sequence;
    String dbpBookName;
    String bookName;  // Localized Bookname from Bible, Used by AudioControlCenter
    int numberOfChapters;

    AudioTOCBook(AudioTOCTestament testament, int index, Cursor cursor) {
        this.testament = testament;
        this.bookId = cursor.getString(0);
        this.bookOrder = cursor.getString(1);
        this.sequence = index;
        this.dbpBookName = cursor.getString(2);
        this.bookName = this.bookId; // Reset by MetaDataReader.readBookNames to bookName
        String chapters = cursor.getString(3);
        try {
            this.numberOfChapters = Integer.parseInt(chapters);
        } catch (Exception ex) {
            this.numberOfChapters = 1;
        }
    }

    //deinit {
    //    print("***** Deinit AudioTOCBook *****")
    //}

    public String toString() {
        String str = "bookId=" + this.bookId +
                ", bookOrder=" + this.bookOrder +
                ", numberOfChapter=" + String.valueOf(this.numberOfChapters);
        return str;
    }

}

File: Plugins/AudioPlayer/src/ios/AudioPlayer/AudioTOCBook.swift
//
//  AudioTOCBook.swift
//  AudioPlayer
//
//  Created by Gary Griswold on 8/8/17.
//  Copyright © 2017 ShortSands. All rights reserved.
//

class AudioTOCBook {
    
    let testament: AudioTOCTestament
    let bookId: String
    let bookOrder: String
    let sequence: Int
    let dbpBookName: String
    var bookName: String // Used by AudioControlCenter
    let numberOfChapters: Int

    init(testament: AudioTOCTestament, index: Int, dbRow: [String?]) {
        self.testament = testament
        self.bookId = dbRow[0]!
        self.bookOrder = dbRow[1]!
        self.sequence = index
        self.dbpBookName = dbRow[2]!
        self.bookName = self.dbpBookName // Reset by MetaDataReader.readBookNames to bookName
        let chapters = dbRow[3]!
        self.numberOfChapters = Int(chapters) ?? 1
    }
    
    deinit {
        print("***** Deinit AudioTOCBook *****")
    }
    
    func toString() -> String {
        let str = "bookId=" + self.bookId +
            ", bookOrder=" + self.bookOrder +
            ", numberOfChapter=" + String(self.numberOfChapters)
        return str
    }
}
